packages/cbioportal-utils/src/ptm/PtmUtils.ts

```diff
@@ -0,0 +1,377 @@
+    Diiodotyrosine: PtmType.Iodination,
+    Triiodothyronine: PtmType.Iodination,
+    Thyroxine: PtmType.Iodination,
+    // Bromination
+    "6'-bromotryptophan": PtmType.Bromination,
+    Bromohistidine: PtmType.Bromination,
+    // Citrullination
+    Citrulline: PtmType.Citrullination,
+    // Lipoylation
+    'N6-lipoyllysine': PtmType.Lipoylation,
+    // Nitration
+    "3'-nitrotyrosine": PtmType.Nitration,
+    'Nitrated tyrosine': PtmType.Nitration,
+    // Oxidation
+    'Cysteine sulfenic acid (-SOH)': PtmType.Oxidation,
+    'Cysteine sulfinic acid (-SO2H)': PtmType.Oxidation,
+    'Cysteine sulfonic acid (-SO3H)': PtmType.Oxidation,
+    'Methionine sulfone': PtmType.Oxidation,
+    'Methionine (R)-sulfoxide': PtmType.Oxidation,
+    'Methionine (S)-sulfoxide': PtmType.Oxidation,
+    'Methionine sulfoxide': PtmType.Oxidation,
+    '(4R)-5-oxoleucine': PtmType.Oxidation,
+    // Pyruvate
+    '2-(S-cysteinyl)pyruvic acid O-phosphothioketal': PtmType.Pyruvate,
+    'N-pyruvate 2-iminyl-cysteine': PtmType.Pyruvate,
+    'N-pyruvate 2-iminyl-valine': PtmType.Pyruvate,
+    'Pyruvic acid (Cys)': PtmType.Pyruvate,
+    'Pyruvic acid (Ser)': PtmType.Pyruvate,
+    'Pyruvic acid (Tyr)': PtmType.Pyruvate,
+    // S-nitrosylation
+    'S-nitrosocysteine': PtmType.SNitrosylation,
+    // Sulfation
+    Sulfoserine: PtmType.Sulfation,
+    Sulfothreonine: PtmType.Sulfation,
+    Sulfotyrosine: PtmType.Sulfation,
+};
+
+export function groupPtmDataByPosition(
+    ptmData: PostTranslationalModification[]
+) {
+    return _.groupBy(ptmData, p => p.residue.start);
+}
+
+export function groupPtmDataByTypeAndPosition(
+    ptmData: PostTranslationalModification[]
+) {
+    const groupedByType = _.groupBy(ptmData, p => p.type);
+
+    const groupedByTypeAndPosition: {
+        [type: string]: { [position: number]: PostTranslationalModification[] };
+    } = {};
+
+    _.keys(groupedByType).forEach(type => {
+        groupedByTypeAndPosition[type] = _.groupBy(
+            groupedByType[type],
+            p => p.residue.start
+        );
+    });
+
+    return groupedByTypeAndPosition;
+}
+
+export function ptmColor(ptms: PostTranslationalModification[]) {
+    let color = PTM_COLORS.default;
+    const uniqueTypes = _.uniq((ptms || []).map(ptm => ptm.type));
+
+    if (uniqueTypes.length === 1) {
+        color = PTM_COLORS[uniqueTypes[0]] || PTM_COLORS.default;
+    } else if (uniqueTypes.length > 1) {
+        color = PTM_COLORS.multiType;
+    }
+
+    return color;
+}
+
+export function compareByPtmTypePriority(a: string, b: string) {
+    const aValue = PTM_PRIORITY[a] || PTM_PRIORITY.default;
+    const bValue = PTM_PRIORITY[b] || PTM_PRIORITY.default;
+
+    if (aValue > bValue) {
+        return 1;
+    } else if (bValue > aValue) {
+        return -1;
+    } else if (a > b) {
+        return 1;
+    } else if (b > a) {
+        return -1;
+    }
+
+    return 0;
+}
+
+function getResiduesFromUniportFeature(ptm: UniprotFeature) {
```

Comment:
looks like these three functions have type on their name "uniport" ?   should it be "uniprot"?